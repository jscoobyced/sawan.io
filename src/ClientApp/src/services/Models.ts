export enum Language {
    English,
    French
}

export interface AllContent {
    navigationMenuContent: NavigationMenuContent;
    homeContent: HomeContent;
}

export interface NavigationMenuContent {
    home: string;
    websiteName: string;
    applications: string;
    cryptoCurrency: string;
    healthMonitor: string;
    about: string;
}

export interface HomeContent {
    articleTitle: string;
    article: string;
}
