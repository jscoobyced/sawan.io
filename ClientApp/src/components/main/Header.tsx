import React from 'react';
import { NavLink } from 'react-router-dom';
import { SignButton } from '../auth/SignButton';
import { HeaderProps } from './Models';

export class Header extends React.Component<HeaderProps> {

    public constructor(props: HeaderProps) {
        super(props);
    }

    public render() {
        return <header className='main'>
            <img src='wings.png' className='logo'
                alt={this.props.navigationMenuContent.websiteName + ' logo'}
                title={this.props.navigationMenuContent.websiteName} />
            <span>{this.props.navigationMenuContent.websiteName}</span>
            <nav>
                <ul>
                    <li><NavLink to={'/'}>
                        {this.props.navigationMenuContent.home}
                    </NavLink>
                    </li>
                    <li>
                        <a href='#' className='menu-space'>&nbsp;</a>
                    </li>
                    <li>
                        <a href='#' className='dropdown'>{this.props.navigationMenuContent.applications}</a>
                        <ul>
                            <li>
                                <NavLink to={'/candle'}>
                                    {this.props.navigationMenuContent.cryptoCurrency}
                                </NavLink>
                            </li>
                            <li>
                                <NavLink to={'/health'}>
                                    {this.props.navigationMenuContent.healthMonitor}
                                </NavLink>
                            </li>
                        </ul>
                    </li>
                    <li>
                        <a href='#' className='dropdown'>{this.props.navigationMenuContent.information}</a>
                        <ul>
                            <li>
                                <NavLink to={'/about'}>
                                    {this.props.navigationMenuContent.about}
                                </NavLink>
                            </li>
                            <li>
                                <NavLink to={'/resume'}>
                                    {this.props.navigationMenuContent.resume}
                                </NavLink>
                            </li>
                            <li>
                                <NavLink to={'/resume/?full'} target='_blank'>
                                    {this.props.navigationMenuContent.resume} <i className='fas fa-print'></i>
                                </NavLink>
                            </li>
                            <li>
                                <NavLink to={'/Cedric Rochefolle - Resume 2019.pdf'} target='_blank'>
                                    {this.props.navigationMenuContent.resume} <i className='fas fa-download'></i>
                                </NavLink>
                            </li>
                        </ul>
                    </li>
                    <li>
                        <SignButton signIn={this.props.signIn} isSignedIn={this.props.isSignedIn}/>
                    </li>
                </ul>
            </nav>
        </header>;
    }
}
