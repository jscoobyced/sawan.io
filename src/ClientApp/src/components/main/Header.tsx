import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../../styles/images/wings.png';
import { HeaderProps } from './Models';

export class Header extends React.Component<HeaderProps> {

    public constructor(props: HeaderProps) {
        super(props);
    }

    public render() {
        return <header>
            <img src={logo} className='logo'
                alt={this.props.navigationMenuContent.websiteName + ' logo'}
                title={this.props.navigationMenuContent.websiteName as string} />
            <span>{this.props.navigationMenuContent.websiteName}</span>
            <nav>
                <ul>
                    <li><NavLink to={'/'}>
                        {this.props.navigationMenuContent.home}
                    </NavLink>
                    </li>
                    <li>
                        <a href='#' className='menu-space'>&nbsp;</a>
                    </li>
                    <li>
                        <a href='#' className='dropdown'>{this.props.navigationMenuContent.applications}</a>
                        <ul>
                            <li>
                                <NavLink to={'/candle'}>
                                    {this.props.navigationMenuContent.cryptoCurrency}
                                </NavLink>
                            </li>
                            <li>
                                <NavLink to={'/health'}>
                                    {this.props.navigationMenuContent.healthMonitor}
                                </NavLink>
                            </li>
                        </ul>
                    </li>
                    <li>
                        <NavLink to={'/'}>
                            {this.props.navigationMenuContent.about}
                        </NavLink>
                    </li>
                </ul>
            </nav>
        </header>;
    }
}
