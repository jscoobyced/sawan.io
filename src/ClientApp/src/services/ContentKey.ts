export class ContentKey {

    // General
    public static WebsiteName = 'WEBSITENAME';

    // Navigation Menu
    public static Home = 'HOME';
    public static About = 'ABOUT';
    public static Applications = 'APPLICATIONS';
    public static CryptoCurrency = 'CRYPTOCURRENCY';
    public static HealthMonitor = 'HEALTHMONITOR';

    // Home
    public static MainArticleTitle = 'MAINARTICLETITLE';
    public static MainArticle = 'MAINARTICLE';

    // Footer
    public static FooterCopyright = 'FOOTERCOPYRIGHT';
    public static FooterCredits = 'FOOTERCREDITS';
    public static FooterYear = 'FOOTERYEAR';
}
