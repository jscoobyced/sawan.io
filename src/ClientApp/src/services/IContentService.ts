import { BlogPage, Language, MainContent } from './Models';

export interface IContentService {
    getMainContent(language: Language): Promise<MainContent>;
    getDefaultMainContent(): MainContent;
    getBlogPage(maxResult: number, from: number): Promise<BlogPage>;
    getDefaultBlogPage(): BlogPage;
}
