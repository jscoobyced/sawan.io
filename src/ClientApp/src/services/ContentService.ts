import { ContentKey } from './ContentKey';
import { AllContent, Language } from './Models';

export class ContentService {

    private readonly content: Array<{ [key: string]: string; }> = [];

    public updateContent(language: Language): AllContent {
        this.content[language] = {};
        this.content[language][ContentKey.WebsiteName] = 'sawan.io';
        this.content[language][ContentKey.Home] = 'Home';
        this.content[language][ContentKey.About] = 'About';
        this.content[language][ContentKey.Applications] = 'Applications';
        this.content[language][ContentKey.CryptoCurrency] = 'Cryptocurrency';
        this.content[language][ContentKey.HealthMonitor] = 'Health Monitor';
        const date = '2018-11-15';
        const displayDate = (new Date(date).toISOString().substring(0, 10));
        this.content[language][ContentKey.MainArticleTitle] = `Latest news - ${displayDate}`;
        this.content[language][ContentKey.MainArticle] = 'Grand opening of sawan.io.<br />This website is about my'
            + ' personal training on various web-technologies: TypeScript, webpack, ReactJS, Secure HTTP headers...'
            + ' It includes a simple (and a bit naive :) ) crypto-currency analyser: it can suggest you to sell'
            + ' or buy. Use at your own risk, it is amateur predictions, you have been warned :)'
            + '<br />There will be also an application to track health metrics (weight, % body fat, vascualar'
            + ' fat ratio... Or not, this will be my next project :)<br /> This project is open-sourced on <a'
            + ' href=\'https://github.com/jscoobyced/sawan.io\'>Github</a>. Enjoy and have fun.';

        return this.updateAllContent(language);
    }

    public defaultContent(): AllContent {
        return {
            navigationMenuContent: {
                about: '',
                applications: '',
                cryptoCurrency: '',
                healthMonitor: '',
                home: '',
                websiteName: '',
            },
            homeContent: {
                article: '',
                articleTitle: ''
            }
        };

    }

    private updateAllContent(language: Language): AllContent {
        return {
            navigationMenuContent: {
                about: this.content[language][ContentKey.About],
                applications: this.content[language][ContentKey.Applications],
                cryptoCurrency: this.content[language][ContentKey.CryptoCurrency],
                healthMonitor: this.content[language][ContentKey.HealthMonitor],
                home: this.content[language][ContentKey.Home],
                websiteName: this.content[language][ContentKey.WebsiteName],
            },
            homeContent: {
                article: this.content[language][ContentKey.MainArticle],
                articleTitle: this.content[language][ContentKey.MainArticleTitle]
            }
        };
    }
}
