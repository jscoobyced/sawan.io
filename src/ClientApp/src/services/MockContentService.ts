import { ContentService } from './ContentService';
import { BlogElement, BlogPage, Language, MainContent } from './Models';

export class MockContentService extends ContentService {

    private readonly data = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do'
        + ' ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco'
        + ' aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit '
        + ' cillum dolore eu fugiat nulla pariatur.<br />Excepteur sint occaecat cupidatat non proident, '
        + ' culpa qui officia deserunt mollit anim id est laborum. Commodo qui imperdiet massa tincidunt.'
        + ' Non enim praesent elementum facilisis leo. Adipiscing diam donec adipiscing tristique. Velit'
        + ' euismod in pellentesque massa placerat. Interdum velit euismod in pellentesque. Phasellus'
        + ' vestibulum lorem sed risus ultricies tristique nulla. Egestas erat imperdiet sed euismod nisi'
        + ' porta lorem mollis aliquam. Nulla facilisi etiam dignissim diam quis enim lobortis scelerisque'
        + ' fermentum. Morbi tristique senectus et netus et malesuada fames.<br />Vitae suscipit tellus'
        + ' mauris a diam maecenas. Sed risus ultricies tristique nulla aliquet. Sagittis eu volutpat odio'
        + ' facilisis mauris sit amet. Tempor orci eu lobortis elementum nibh tellus molestie nunc non.'
        + ' Turpis nunc eget lorem dolor.';

    protected updateMainContent(language: Language): Promise<MainContent> {
        const content: MainContent = {
            menuContent: {
                links: []
            },
            navigationMenuContent: {
                about: 'About',
                applications: 'Applications',
                cryptoCurrency: 'Crypto-Currency',
                healthMonitor: 'Health Monitor',
                home: 'Home',
                websiteName: 'sawan.io'
            },
            footerContent: {
                copyright: 'Copyright',
                credits: 'sawan.io',
                year: '2018 - ' + new Date().getFullYear()
            }
        };
        return Promise.resolve(content);
    }

    protected updateBlogPage(maxResult: number): Promise<BlogPage> {
        const date = '2018-11-15';
        const displayDate = (new Date(date).toISOString().substring(0, 10));
        const articles: BlogElement[] = [
            {
                article: `Latest news - ${displayDate}`,
                articleTitle: 'Grand opening of sawan.io.<br />This website is about my'
                    + ' personal training on various web-technologies: TypeScript, webpack, ReactJS, Secure HTTP'
                    + '  headers... It includes a simple (and a bit naive) crypto-currency analyser: it can'
                    + ' suggest you to sell or buy. Use at your own risk, it is amateur predictions, you have'
                    + ' been warned.<br />There will be also an application to track health metrics (weight,'
                    + ' % body fat, vascular fat ratio...). Or maybe not, this might be my next project.<br />'
                    + ' This project is open-sourced on <atarget=\'_blank\''
                    + ' href=\'https://github.com/jscoobyced/sawan.io\'>Github</a>. Enjoy and have fun.'
            },
            {
                article: 'Lorem Ipsum',
                articleTitle: this.data
            },
            {
                article: 'Some other news',
                articleTitle: this.data
            }
        ];
        return Promise.resolve({ articles });
    }
}
