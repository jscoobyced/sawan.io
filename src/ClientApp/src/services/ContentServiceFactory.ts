import { ContentService } from './ContentService';
import { IContentService } from './IContentService';
import { MockContentService } from './MockContentService';

export class ContentServiceFactory {

    public static GetContentService(mode: string): IContentService {
        if (mode === 'production') {
            return new ContentService();
        }

        return new MockContentService();
    }
}