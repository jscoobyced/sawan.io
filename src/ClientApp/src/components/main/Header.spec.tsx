import { shallow } from 'enzyme';
import React from 'react';
import { ContentService } from '../../services/ContentService';
import { Header } from './Header';

test('Header component is unchanged.', () => {
    const navigationMenuContent = new ContentService().getDefaultMainContent().navigationMenuContent;
    const header = shallow(<Header navigationMenuContent={navigationMenuContent} />);

    expect(header).toMatchSnapshot();
});