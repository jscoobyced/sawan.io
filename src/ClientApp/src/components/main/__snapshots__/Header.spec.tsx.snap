// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component is unchanged. 1`] = `
<header>
  <img
    alt=" logo"
    className="logo"
    src="/dist/images/wings.svg"
    title=""
  />
  <span />
  <nav>
    <ul>
      <li>
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        />
      </li>
      <li>
        <a
          className="menu-space"
          href="#"
        >
          Â 
        </a>
      </li>
      <li>
        <a
          className="dropdown"
          href="#"
        />
        <ul>
          <li>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/candle"
            />
          </li>
          <li>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/health"
            />
          </li>
        </ul>
      </li>
      <li>
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        />
      </li>
    </ul>
  </nav>
</header>
`;
